pipeline-yaml_5b5744d40428631d7a8940a9.md

```diff
@@ -615,6 +614,53 @@ were also defined on the `task` level.
 	          - echo $VAR_1
 	          - echo $APP_ENV
 
+### matrix
+
+The `martix` property allows you to define one or more environment variables
+with multiple values. After that, it creates multiple jobs by calculating the
+Cartesian product of all the sets of user defined values.
+
+So, the final outcome of the `matrix` property is concurrent jobs with the
+commands that are defined in the respective `jobs` property but different
+environment variables that are defined in the `matrix` property.
+
+#### Example of matrix
+
+The following pipeline illustrates the use of the `matrix` property:
+
+	version: v1.0
+	name: Using the matrix property
+	agent:
+	  machine:
+	    type: e1-standard-2
+	    os_image: ubuntu1804
+    
+	blocks:
+	  - name: Elixir + Erlang
+	    task:
+	        jobs:
+	        - name: Elixir + Erlang matrix
+	          commands:
+	            - echo $ELIXIR
+	            - echo $ERLANG
+	          matrix:
+	            - env_var: ELIXIR
+	              values: ["1.3", "1.4"]
+	            - env_var: ERLANG
+	              values: ["19", "20", "21"]
+
+This will create six jobs because there are 6 elements in the Cartesian product
```

Comment:
Let's not repeat the mathematical term nobody remembers. :)

We can simplify both paragraphs to one sentence: _In this example the `matrix` expands to 6 parallel jobs as there are 2 * 3 = 6 combinations of provided environment variables:_ (followed by the list)

Reply:
Good point.